### PR TITLE
Adjust configuration and extend tree-shaking support

### PR DESCRIPTION
**Pull request by rryter**
For review purposes it's best to look at the commits individually.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -53,7 +53,7 @@ const mySchema = [
   },
 ];
 
-let myERC725 = new ERC725(schema[, address, provider, ipfsGateway])
```
Comment (frozeman) [inline README.md:56]: i guess we want to add here an example config object and also use the variables from above.

best move line 28 and 27 below 54 you

File: src/index.ts [hunk 2]
```diff
@@ -24,7 +24,9 @@ import axios from 'axios'
 import GraphSource from './providers/subgraphProviderWrapper'
 import Web3Source from './providers/web3ProviderWrapper'
 import EthereumSource from './providers/ethereumProviderWrapper'
-import { utils } from './lib/utils'
+import {
+    decodeAllData, decodeKey, decodeKeyValue, encodeAllData, encodeArrayKey, encodeKey, getSchemaElement
+} from './lib/utils'
```
Comment (frozeman) [inline src/index.ts:29]: Isn't the libs just an internal library? I think selecting here functions doesnt improve a lot. External dependencies are the problem

**Review by frozeman — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 2 above)
Comment (rryter) [reply src/index.ts:29]: I assumed the utils would also be imported directly from time to time. Either way it doesnt hurt imo.

I'm setting up a little project to investigate the size properly.

File: README.md (hunk 1 above)
Comment (rryter) [reply README.md:56]: Yes, very good input.

**Pull request merged**